### PR TITLE
coprocess_python: use defer C.free for all C strings

### LINKED ISSUE #1228: Possible Leak
https://github.com/TykTechnologies/tyk/blob/c607854b9d50d40b79695e9f60a50c47a356e3f8/coprocess_python.go#L247-L254

Should the IF in line 247 evaluate true because of any reason. The release memory a few lines below wont execute, that will mean a Leak.

Also this happens many different times in the Coprocess and C integration. The correct way to use this release methods is by using a `defer C.free(....`

### PR DESCRIPTION
**Pull request by matiasinsaurralde**
Fixes #1228.

### CODE REVIEW

File: coprocess_python.go [hunk 1]
```diff
@@ -213,8 +213,8 @@ func (d *PythonDispatcher) Reload() {
 func (d *PythonDispatcher) HandleMiddlewareCache(b *apidef.BundleManifest, basePath string) {
 	go func() {
 		runtime.LockOSThread()
-		defer runtime.UnlockOSThread()
```
Comment (mvdan) [inline coprocess_python.go]: is this on purpose? are you sure it's safe to remove?

File: coprocess_python.go [hunk 2]
```diff
@@ -240,27 +240,25 @@ func PythonLoadDispatcher() error {
 // PythonNewDispatcher creates an instance of TykDispatcher.
 func PythonNewDispatcher(middlewarePath, eventHandlerPath string, bundlePaths []string) (coprocess.Dispatcher, error) {
 	CMiddlewarePath := C.CString(middlewarePath)
+	defer C.free(unsafe.Pointer(CMiddlewarePath))
 	CEventHandlerPath := C.CString(eventHandlerPath)
+	defer C.free(unsafe.Pointer(CEventHandlerPath))
 	CBundlePaths := C.CString(strings.Join(bundlePaths, ":"))
+	defer C.free(unsafe.Pointer(CBundlePaths))
```
Comment (mvdan) [inline coprocess_python.go]: We had an actual leak here as there were only two frees before, correct?

**Comment by mvdan** [comment]
Also, use "Fixes #n" to actually close the issue once merged.

File: coprocess_python.go (hunk 1 above)
Comment (matiasinsaurralde) [reply coprocess_python.go]: @mvdan I've updated it with a `func` that performs both operations:`C.free` and `runtime.UnlockOSThread`.

File: coprocess_python.go (hunk 2 above)
Comment (matiasinsaurralde) [reply coprocess_python.go]: That's right, this is the missing one.

**Comment by matiasinsaurralde** [comment]
Just a quick note: if #1260 gets merged, this won't be necessary at all.

**Comment by buger** [comment]
Closing in favor of #1260

**Pull request closed without merging**